codegen/src/Logger.scala

```diff
@@ -1,15 +1,54 @@
 package besom.codegen
 
+import besom.codegen.Logger.Level
+import besom.codegen.Logger.Level.Info
+
 import scala.collection.mutable.ListBuffer
 
-class Logger {
-  private val buffer = ListBuffer.empty[String]
+class Logger(val printLevel: Level = Info) {
+  import Logger.Level._
+
+  private val buffer     = ListBuffer.empty[String]
+  private var errorCount = 0
```

Comment:
It might be good to print these numbers in the log file and/or directly to the console to make it easier to assess the severity of problems for a given provider